hooks/tk-multi-publish2/basic/publish_session_usd.py

```diff
@@ -95,6 +115,148 @@ def item_filters(self):
         """
         return ["maya.session.usd"]
 
+    def create_settings_widget(self, parent):
+        """
+        Creates a custom widget for the given parent widget.
+
+        :param parent: Parent widget to host the custom UI.
+        :return: The QWidget containing the custom UI.
+        """
+        from sgtk.platform.qt import QtGui
+
+        usd_publish_menu = QtGui.QGroupBox(parent)
+        usd_publish_menu.setTitle("USD publishing")
+        menu_layout = QtGui.QVBoxLayout()
+        menu_layout.addStretch()
+
+        # Description text
+        description_label = QtGui.QLabel(self.description)
+        description_label.setWordWrap(True)
+        description_label.setOpenExternalLinks(True)
+        menu_layout.addWidget(description_label)
+
+        # Active selection
+        active_selection_checkbox = QtGui.QCheckBox("Export active selection only")
+        menu_layout.addWidget(active_selection_checkbox)
+
+        # Disable animations
+        disable_animation_checkbox = QtGui.QCheckBox("Disable animations")
+        menu_layout.addWidget(disable_animation_checkbox)
+
+        # Current frame
+        current_frame_checkbox = QtGui.QCheckBox("Export current frame only")
+        menu_layout.addWidget(current_frame_checkbox)
+
+        usd_publish_menu.setLayout(menu_layout)
+
+        return usd_publish_menu
+
+    def get_ui_settings(self, widget, items=None):
+        """
+        Invoked by the Publisher when the selection changes. This method gathers the settings
+        on the previously selected task, so that they can be later used to repopulate the
+        custom UI if the task gets selected again. They will also be passed to the accept, validate,
+        publish and finalize methods, so that the settings can be used to drive the publish process.
+
+        The widget argument is the widget that was previously created by
+        `create_settings_widget`.
+
+        The method returns a dictionary, where the key is the name of a
+        setting that should be updated and the value is the new value of that
+        setting. Note that it is up to you how you want to store the UI's state as
+        settings and you don't have to necessarily to return all the values from
+        the UI. This is to allow the publisher to update a subset of settings
+        when multiple tasks have been selected.
+
+        Example::
+
+            {
+                 "setting_a": "/path/to/a/file"
+            }
+
+        :param widget: The widget that was created by `create_settings_widget`
+        """
+        from sgtk.platform.qt import QtGui
+
+        checkbox_settings_list = widget.findChildren(QtGui.QCheckBox)
+
+        for checkbox in checkbox_settings_list:
```

Comment:
Wouldn't it be easier to assign the checkboxes to local parameters and reference them with self instead of looping through them all?